demo/csrc/object_detection.cpp

```diff
@@ -37,12 +37,35 @@ int main(int argc, char *argv[]) {
     return 1;
   }
 
-  fprintf(stderr, "bbox_count=%d\n", *res_count);
+  fprintf(stdout, "bbox_count=%d\n", *res_count);
 
   for (int i = 0; i < *res_count; ++i) {
     const auto &box = bboxes[i].bbox;
-    fprintf(stderr, "box %d, left=%.2f, top=%.2f, right=%.2f, bottom=%.2f, label=%d, score=%.4f\n",
+    const auto &mask = bboxes[i].mask;
+
+    fprintf(stdout, "box %d, left=%.2f, top=%.2f, right=%.2f, bottom=%.2f, label=%d, score=%.4f\n",
             i, box.left, box.top, box.right, box.bottom, bboxes[i].label_id, bboxes[i].score);
+
+    // skip detections with invalid bbox coordinates
+    if (box.left < 1 && box.top < 1) {
+      continue;
+    }
+
+    // generate mask overlay if model exports masks
+    if (mask != nullptr) {
+      fprintf(stdout, "mask %d, height=%d, width=%d\n", i, mask->height, mask->width);
+
+      cv::Mat imgMask(mask->height, mask->width, CV_8UC1, &mask->data[0]);
+      cv::Rect roi((int)box.left - 1, (int)box.top - 1, mask->width, mask->height);
```

Comment:
The current logic of the post-processing of instance segmentation is like 
```C++
auto x0 = std::max(std::floor(bbox[0]) - 1, 0.f);
auto y0 = std::max(std::floor(bbox[1]) - 1, 0.f);
```
It is preferred to keep the same logic when recovering the whole-image mask so that the benchmark results would stay the same.

Also, the top-left corner of invalid boxes (either FP or dummy outputs from backends) is not well defined and not reliable  for box filtering. In this case, edge length or area thresholding is can used in combination with score thresholding to get better visual results.

Reply:
@lzhangzz So if I understand correctly, we should do something like:

```cpp
auto x0 = std::max(std::floor(box.left) - 1, 0.f);
auto y0 = std::max(std::floor(box.top) - 1, 0.f);
cv::Rect roi((int) x0, (int) y0, mask->width, mask->height);
```

Reply:
> ```c++
> auto x0 = std::max(std::floor(box.left) - 1, 0.f);
> auto y0 = std::max(std::floor(box.top) - 1, 0.f);
> cv::Rect roi((int) x0, (int) y0, mask->width, mask->height);
> ```
Yes

> And for filtering the invalid boxes, we should look at not just the top-left but the side or area, in case it is zero, indicating an invalid bbox?

Don't use the top-left coordinate, just size/area and score. 

And actually there is a "min_bbox_size" property for detection models, however it is default to 0 in most model configs. So for demostration & visualization purpose, we can perform the filtering again on the detections returned by API.

Reply:
@lzhangzz OK, I have updated it now according to your suggestions.